cmd/storage-rest-server.go

```diff
@@ -179,6 +179,33 @@ func (s *storageRESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Req
 	}
 }
 
+// WriteAllHandler - write to file all content.
+func (s *storageRESTServer) WriteAllHandler(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		return
+	}
+	vars := mux.Vars(r)
+	volume := vars[storageRESTVolume]
+	filePath := vars[storageRESTFilePath]
+
+	if r.ContentLength < 0 {
+		s.writeErrorResponse(w, errInvalidArgument)
+		return
+	}
+
+	buf := make([]byte, r.ContentLength)
+	_, err := io.ReadFull(r.Body, buf)
```

Comment:
I think it would be good to check for read length here and compare it with r.ContentLength

Reply:
It will never happen @vadmeste r.Body is limited to ContentLength internally. It would be a frivolous check.

Reply:
> It will never happen @vadmeste r.Body is limited to ContentLength internally. It would be a frivolous check.

Yes right, this test confimrs that r.Body returns an error when body is short https://golang.org/src/net/http/response_test.go#L786